src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -596,7 +597,9 @@ private bool TakeStartLine(SocketInput input)
                 {
                     return false;
                 }
-                var requestUri = begin.GetString(scan);
+
+                var beginPath = begin;
+                var endPath = scan;
```

Comment:
`+` is not a valid uri encoding for space, it's primarily used in html encoding and sometimes in queries.